targets/TARGET_STM/can_api.c

```diff
@@ -1177,6 +1184,11 @@ static void can_irq(CANName name, int id)
     tmp1 = __HAL_CAN_MSG_PENDING(&CanHandle, CAN_FIFO0);
     tmp2 = __HAL_CAN_GET_IT_SOURCE(&CanHandle, CAN_IT_FMP0);
 
+    // In legacy can, reading is the only way to clear rx interrupt. But can_read has mutex locks
```

Comment:
"legacy" is compared to FDCAN, or ST internal HAL_CAN_LEGACY_MODULE_ENABLED ?

Reply:
Updated the code as per the comment.